### PR TITLE
feat!: `byteArray` in Data Contracts schema

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Integrate `byteArray` keyword (#245) to Data Contract schema instead of ugly `contentEncoding`. 

## What was done?

`byteArray` defined in data contract meta schema. `contentEncoding` keyword is restricted.
`DataContract#getEncodedProperties` replaced with `DataContract#getBinaryProperties`

## How Has This Been Tested?

With tests

## Breaking Changes

`contentEncoding` is restricted.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by antouhou — APPROVED** [review]
utACK

**Review by jawid-h — APPROVED** [review]
Okie 👍

File: test/unit/document/Document.spec.js [hunk 1]
```diff
@@ -329,36 +328,36 @@ describe('Document', () => {
...
 
-      lodashGetMock.returns(value.base64Value);
+      lodashGetMock.returns(value.byteArrayProperty);
 
       const result = document.set(path, value);
 
       expect(result).to.equal(document);
 
-      const valueToSet = new EncodedBuffer(value.base64Value, 'base64');
+      const valueToSet = new EncodedBuffer(value.byteArrayProperty, 'base64');
```
Comment (shuplenkov) [inline test/unit/document/Document.spec.js:353]: new EncodedBuffer(value.byteArrayProperty, EncodedBuffer.ENCODING.BASE64);

Comment (shumkov) [reply test/unit/document/Document.spec.js:353]: Let's keep it as it is for now. we will remove EncodedBuffer in the next PR

**Pull request merged**